### PR TITLE
Restructure code into packages

### PR DESCRIPTION
**Pull request by parkr**
Hey @benbalter! This is a big one, so let me explain:

1. I created a `config/` package. This package provides an interface type called `Provider`. It currently has one implementation, `FileProvider`, which reads from `~/.config/google`. In a subsequent PR, I want to add OS X keychain support as a separate provider.
2. I moved any non-interactive logic to `zoom.go` and `client.go` in the base. Things like setting up the calendar service and anything else related to querying Google, parsing zoom links, etc. All the other "domain logic" that didn't make sense in `config/`.
3. `cmd/zoom/main.go` handles 100% of the interaction with the user. All printing to stdout, all reading, all opening of links, etc. It also handles missing `client_secrets.json` and `token.json` files and helps the user set those up.
4. The command now has an `-import` flag. You give it a file path to your client secrets JSON file and it'll import and store it before continuing on. If I have coded it correctly, a brand new user should run `zoom`, follow the instructions to create the Google App, then run `zoom -import=$HOME/Downloads/blah.json`, then follow the instructions to authorize, then it should continue and open their meetings. It's not very _few_ steps, but it works.

That's all! Not so terrible. Next up, I want to write better documentation and tests for these improvements. Documentation may come in this PR, but tests likely won't. 😄 In such a big diff, it'll be easier to review with tests by themselves. Maybe I'll open a PR for each new test file. 🙂

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	calendar "google.golang.org/api/calendar/v3"
+)
+
+// NewGoogleClient creates a new client using the token from the given provider.
+func NewGoogleClient(provider config.Provider) (*http.Client, error) {
+	conf, err := provider.GoogleClientConfig()
+	if err != nil {
+		return nil, err
```
Comment (benbalter) [inline client.go:17]: TIL'd go uses tabs.

File: cmd/zoom/main.go [hunk 2]
```diff
@@ -1,199 +1,128 @@
...
+2. Grant the project Calendar API access
+	1. Click "Enable API"
+	2. Type "Calendar" in the search box
+	3. Click "Calendar API"
+	4. Click "Enable"
+3. Grab your credentials
+	1. Click "Credentials" on the left side
+	2. Create a new OAuth credential with type "other"
+	3. Download the credential to ~/.config/google/client_secrets.json (icon, right side)
+4. Run 'zoom -import=Downloads/client_secrets.json' and follow the instructions to authorize the app.
```
Comment (benbalter) [inline cmd/zoom/main.go:33]: :heart:

File: config/file.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
+package config
```
Comment (benbalter) [inline config/file.go:1]: does the filename need to be e.g., `file_provider`?

**Review by benbalter — APPROVED** [review]
Wow. Sparkles. This is great. Very elegant abstraction of the provider. Thanks for taking the time to make this tiny app 10x better.

**Comment by benbalter** [comment]
> I want to add OS X keychain support as a separate provider.

All for that. IIRC, the Go implementation was _intended_ to import from the Ruby implementation. Would be cool if the Keychain implementation could transparently import from either legacy file implementation.

File: config/file.go (hunk 3 above)
Comment (parkr) [reply config/file.go:1]: Not strictly. All .go files in a package are pulled into the build process and their filenames have no special meaning for `go build` or `go install` or `go get`. Files ending in `_test.go` have a special meaning for `go test`, but that's the only instance I can think of at the moment. 😄

Comment (parkr) [reply config/file.go:1]: If you would like it to be file_provider.go, I'm happy to rename it!

**Comment by parkr** [comment]
> the Go implementation was intended to import from the Ruby implementation. Would be cool if the Keychain implementation could transparently import from either legacy file implementation.

Seamless backwards compatibility is a must-have. Maybe an `importers/` package that can take multiple providers and migrate from one to the other? 😄 For another PR...

**Comment by parkr** [comment]
```
Your next meeting is "Design Review", organized by Johnny Appleseed.
It starts 30 minutes from now.
Calendar event URL: https://www.google.com/calendar/event?eid=foobar

No Zoom URL found in the meeting.
```

**Comment by parkr** [comment]
@benbalter Let me know if there is anything else you’d like to see in this PR! :heart:

**Pull request merged**